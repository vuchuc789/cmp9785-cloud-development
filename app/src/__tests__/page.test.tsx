import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Page from '../app/page';

Object.defineProperty(globalThis, 'crypto', {
  value: {
    randomUUID: () => '8eb27645-39d9-4f26-a10a-4bfdcd9cc233',
  },
});

jest.mock('next/navigation', () => ({
  useRouter: () => ({}),
}));

jest.mock('../contexts/auth', () => ({
  useAuth: jest.fn(() => ({ state: { accessToken: 'abc', isLoading: false } })),
}));

describe('Homepage', () => {
  it('renders unchanged', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });
});
