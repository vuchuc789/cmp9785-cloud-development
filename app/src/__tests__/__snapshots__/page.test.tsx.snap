// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders unchanged 1`] = `
<div>
  <div
    class="grid grid-rows-[20px_1fr_20px] items-center justify-items-center grow p-8 pb-20 gap-16 sm:p-20 font-[family-name:var(--font-geist-sans)]"
  >
    <main
      class="flex flex-col gap-8 row-start-2 items-start"
    >
      <div
        class="flex gap-4 flex-wrap"
      >
        <svg
          fill="none"
          height="72"
          role="img"
          viewBox="0 0 80 80"
          width="72"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="80"
            id=":r0:"
            maskUnits="userSpaceOnUse"
            width="80"
            x="0"
            y="0"
          >
            <rect
              fill="#FFFFFF"
              height="80"
              rx="160"
              width="80"
            />
          </mask>
          <g
            mask="url(#:r0:)"
          >
            <rect
              fill="#C20D90"
              height="80"
              width="80"
            />
            <rect
              fill="#92A1C6"
              height="80"
              transform="translate(2 2) rotate(278 40 40)"
              width="80"
              x="10"
              y="30"
            />
            <circle
              cx="40"
              cy="40"
              fill="#146A7C"
              r="16"
              transform="translate(15 -15)"
            />
            <line
              stroke="#F0AB3D"
              stroke-width="2"
              transform="translate(16 16) rotate(196 40 40)"
              x1="0"
              x2="80"
              y1="40"
              y2="40"
            />
          </g>
        </svg>
        <svg
          fill="none"
          height="72"
          role="img"
          viewBox="0 0 36 36"
          width="72"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="36"
            id=":r1:"
            maskUnits="userSpaceOnUse"
            width="36"
            x="0"
            y="0"
          >
            <rect
              fill="#FFFFFF"
              height="36"
              rx="72"
              width="36"
            />
          </mask>
          <g
            mask="url(#:r1:)"
          >
            <rect
              fill="#F0AB3D"
              height="36"
              width="36"
            />
            <rect
              fill="#C20D90"
              height="36"
              rx="6"
              transform="translate(9 -5) rotate(139 18 18) scale(1.1)"
              width="36"
              x="0"
              y="0"
            />
            <g
              transform="translate(4.5 -5) rotate(9 18 18)"
            >
              <path
                d="M15 20c2 1 4 1 6 0"
                fill="none"
                stroke="#FFFFFF"
                stroke-linecap="round"
              />
              <rect
                fill="#FFFFFF"
                height="2"
                rx="1"
                stroke="none"
                width="1.5"
                x="10"
                y="14"
              />
              <rect
                fill="#FFFFFF"
                height="2"
                rx="1"
                stroke="none"
                width="1.5"
                x="24"
                y="14"
              />
            </g>
          </g>
        </svg>
        <svg
          fill="none"
          height="72"
          role="img"
          viewBox="0 0 80 80"
          width="72"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="80"
            id=":r2:"
            maskUnits="userSpaceOnUse"
            width="80"
            x="0"
            y="0"
          >
            <rect
              fill="#FFFFFF"
              height="80"
              rx="160"
              width="80"
            />
          </mask>
          <g
            mask="url(#:r2:)"
          >
            <rect
              fill="#C20D90"
              height="80"
              width="80"
            />
            <path
              d="M32.414 59.35L50.376 70.5H72.5v-71H33.728L26.5 13.381l19.057 27.08L32.414 59.35z"
              fill="#92A1C6"
              filter="url(#filter_:r2:)"
              transform="translate(6 6) rotate(278 40 40) scale(1.3)"
            />
            <path
              d="M22.216 24L0 46.75l14.108 38.129L78 86l-3.081-59.276-22.378 4.005 12.972 20.186-23.35 27.395L22.215 24z"
              fill="#146A7C"
              filter="url(#filter_:r2:)"
              style="mix-blend-mode: overlay;"
              transform="translate(1 -1) rotate(57 40 40) scale(1.3)"
            />
          </g>
          <defs>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              id="filter_:r2:"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <feblend
                in="SourceGraphic"
                in2="BackgroundImageFix"
                result="shape"
              />
              <fegaussianblur
                result="effect1_foregroundBlur"
                stdDeviation="7"
              />
            </filter>
          </defs>
        </svg>
        <svg
          fill="none"
          height="72"
          role="img"
          viewBox="0 0 80 80"
          width="72"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="80"
            id=":r3:"
            mask-type="alpha"
            maskUnits="userSpaceOnUse"
            width="80"
            x="0"
            y="0"
          >
            <rect
              fill="#FFFFFF"
              height="80"
              rx="160"
              width="80"
            />
          </mask>
          <g
            mask="url(#:r3:)"
          >
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              x="20"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              x="40"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="60"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="10"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              x="30"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="50"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="70"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              y="10"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              y="20"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              y="30"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              y="40"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              y="50"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              y="60"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              y="70"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="20"
              y="10"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="20"
              y="20"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="20"
              y="30"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="20"
              y="40"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="20"
              y="50"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="20"
              y="60"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              x="20"
              y="70"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              x="40"
              y="10"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="40"
              y="20"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="40"
              y="30"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="40"
              y="40"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="40"
              y="50"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="40"
              y="60"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="40"
              y="70"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="60"
              y="10"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              x="60"
              y="20"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="60"
              y="30"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              x="60"
              y="40"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="60"
              y="50"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="60"
              y="60"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              x="60"
              y="70"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="10"
              y="10"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="10"
              y="20"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="10"
              y="30"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="10"
              y="40"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="10"
              y="50"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="10"
              y="60"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="10"
              y="70"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="30"
              y="10"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="30"
              y="20"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              x="30"
              y="30"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="30"
              y="40"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="30"
              y="50"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              x="30"
              y="60"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="30"
              y="70"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              x="50"
              y="10"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="50"
              y="20"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              x="50"
              y="30"
            />
            <rect
              fill="#146A7C"
              height="10"
              width="10"
              x="50"
              y="40"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="50"
              y="50"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="50"
              y="60"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="50"
              y="70"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="70"
              y="10"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              x="70"
              y="20"
            />
            <rect
              fill="#C20D90"
              height="10"
              width="10"
              x="70"
              y="30"
            />
            <rect
              fill="#C271B4"
              height="10"
              width="10"
              x="70"
              y="40"
            />
            <rect
              fill="#F0AB3D"
              height="10"
              width="10"
              x="70"
              y="50"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="70"
              y="60"
            />
            <rect
              fill="#92A1C6"
              height="10"
              width="10"
              x="70"
              y="70"
            />
          </g>
        </svg>
        <svg
          fill="none"
          height="72"
          role="img"
          viewBox="0 0 90 90"
          width="72"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="90"
            id=":r4:"
            maskUnits="userSpaceOnUse"
            width="90"
            x="0"
            y="0"
          >
            <rect
              fill="#FFFFFF"
              height="90"
              rx="180"
              width="90"
            />
          </mask>
          <g
            mask="url(#:r4:)"
          >
            <path
              d="M0 0h90v45H0z"
              fill="#C20D90"
            />
            <path
              d="M0 45h90v45H0z"
              fill="#92A1C6"
            />
            <path
              d="M83 45a38 38 0 00-76 0h76z"
              fill="#92A1C6"
            />
            <path
              d="M83 45a38 38 0 01-76 0h76z"
              fill="#146A7C"
            />
            <path
              d="M77 45a32 32 0 10-64 0h64z"
              fill="#146A7C"
            />
            <path
              d="M77 45a32 32 0 11-64 0h64z"
              fill="#F0AB3D"
            />
            <path
              d="M71 45a26 26 0 00-52 0h52z"
              fill="#F0AB3D"
            />
            <path
              d="M71 45a26 26 0 01-52 0h52z"
              fill="#C20D90"
            />
            <circle
              cx="45"
              cy="45"
              fill="#C271B4"
              r="23"
            />
          </g>
        </svg>
        <svg
          fill="none"
          height="72"
          role="img"
          viewBox="0 0 80 80"
          width="72"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            height="80"
            id=":r5:"
            maskUnits="userSpaceOnUse"
            width="80"
            x="0"
            y="0"
          >
            <rect
              fill="#FFFFFF"
              height="80"
              rx="160"
              width="80"
            />
          </mask>
          <g
            mask="url(#:r5:)"
          >
            <path
              d="M0 0h80v40H0z"
              fill="url(#gradient_paint0_linear_8eb27645-39d9-4f26-a10a-4bfdcd9cc233)"
            />
            <path
              d="M0 40h80v40H0z"
              fill="url(#gradient_paint1_linear_8eb27645-39d9-4f26-a10a-4bfdcd9cc233)"
            />
          </g>
          <defs>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="gradient_paint0_linear_8eb27645-39d9-4f26-a10a-4bfdcd9cc233"
              x1="40"
              x2="40"
              y1="0"
              y2="40"
            >
              <stop
                stop-color="#C20D90"
              />
              <stop
                offset="1"
                stop-color="#92A1C6"
              />
            </lineargradient>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="gradient_paint1_linear_8eb27645-39d9-4f26-a10a-4bfdcd9cc233"
              x1="40"
              x2="40"
              y1="40"
              y2="80"
            >
              <stop
                stop-color="#146A7C"
              />
              <stop
                offset="1"
                stop-color="#F0AB3D"
              />
            </lineargradient>
          </defs>
        </svg>
      </div>
      <h1
        class="scroll-m-20 text-4xl font-extrabold tracking-tight lg:text-5xl"
      >
        Explore millions of creative works
      </h1>
      <p
        class="leading-7 [&:not(:first-child)]:mt-6"
      >
        An extensive library of free stock photos, images, and audio, available for free use.
      </p>
      <div
        class="flex gap-4 items-start flex-col sm:flex-row"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-10 px-6 has-[>svg]:px-4 rounded-full cursor-pointer"
          data-slot="button"
        >
          <svg
            class="lucide lucide-search"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <path
              d="m21 21-4.3-4.3"
            />
          </svg>
          Search now
        </button>
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-[color,box-shadow] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border border-input bg-background shadow-xs hover:bg-accent hover:text-accent-foreground h-10 px-6 has-[>svg]:px-4 rounded-full cursor-pointer"
          data-slot="button"
        >
          <svg
            class="lucide lucide-log-in"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 3h4a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-4"
            />
            <polyline
              points="10 17 15 12 10 7"
            />
            <line
              x1="15"
              x2="3"
              y1="12"
              y2="12"
            />
          </svg>
          Or login if you haven't...
        </button>
      </div>
    </main>
  </div>
</div>
`;
